Use user profile image instead of gravatar when available

GordonBeeming opened the pull request:
## ☕️ Reasoning

Users wanted to see their GitButler profile image instead of the gravatar one when they've suppled one

## 🧢 Changes

Updated locations where gravatarUrl were referenced for various git artifacts to check the users email, and if it's a match, then use the users profile

## 🎫 Affected issues

Fixes: #2589

**NOTE:** I've never worked with this tech stack, but am trying to learn... and really want to contribute to GitButler, so starting small.

If I've completely missed the mark on how to implement this change, please let me know and I'll do some more learning 😅

GordonBeeming committed:
feat: Use user profile image instead of gravatar when available 

fixes #2589

https://github.com/gitbutlerapp/gitbutler/issues/2589

estib-vega commented on apps/desktop/src/lib/navigation/BranchListingSidebarEntry.svelte:
```diff
@@ -92,15 +96,18 @@
 			const name = (await gitConfigService.get('user.name')) || unknownName;
 			const email = (await gitConfigService.get('user.email')) || unknownEmail;
-			const srcUrl = await gravatarUrlFromEmail(email);
+			const srcUrl = $user?.picture ? $user?.picture : await gravatarUrlFromEmail(email);
```
nit: Can we make this a bit more concise, like so:
```ts
$user?.picture ?? (await gravatarUrlFromEmail(email))
```

Also, I guess here we also need to check whether the `$user` email matches the git config's `user.email`, right?

GordonBeeming replied:
Originally my thoughts were that because it's checking the gitconfig, at this point it's definitely "you", so use your profile image. 
However for consistency it does make sense to check the email 🙂

estib-vega replied:
It's a good point.
At least for now, I think it makes sense to compare against the email always so that the gitbutler's account picture appears only when using a git profile that matches, and a different picture if you decide to use another git profile (in the case you use different emails for different projects, for example).

GordonBeeming replied:
100% agree - I have this that config locally 😅 I think my GB email is different to both my normal git emails (work and personal) 🙂

estib-vega commented on apps/desktop/src/lib/components/EditMode.svelte:
```diff
@@ -54,6 +59,15 @@
 		});
 	});
 
+	$effect(() => {
+		if (commit) {
+			authorImgUrl =
+				commit.author.email?.toLowerCase() === $user?.email?.toLowerCase()
+					? $user?.picture
+					: commit.author.gravatarUrl;
+		}
+	});
+
```
Instead of setting this from inside an effect, let's create a derived value, please:
```ts
const authorImgUrl = $derived.by(() => {
    if (commit) {
	return commit.author.email?.toLowerCase() === $user?.email?.toLowerCase()
		? $user?.picture
		: commit.author.gravatarUrl;
    }
    return undefined;
});
```
We want to avoid setting state from inside effects if possible :)

estib-vega commented on apps/desktop/src/lib/commit/CommitCard.svelte:
```diff
@@ -133,6 +137,10 @@
 	}
 
 	const commitShortSha = commit.id.substring(0, 7);
+	const authorImgUrl =
+		commit.author.email?.toLowerCase() === $user?.email?.toLowerCase()
+			? $user?.picture
+			: commit.author.gravatarUrl;
```
Ideally, this should be also a `$derived` value as well. 
Please, check out my comment below.

estib-vega commented on apps/desktop/src/lib/navigation/PullRequestSidebarEntry.svelte:
```diff
@@ -15,6 +16,14 @@
+	const userService = getContext(UserService);
+	const user = userService.user;
+
+	const authorImgUrl =
+		pullRequest.author?.email?.toLowerCase() === $user?.email?.toLowerCase()
+			? $user?.picture
+			: pullRequest.author?.gravatarUrl;
+
```
Same as the comment above, let's turn this into a `$derived` value, please

estib-vega reviewed (changes requested):
Thank you for your contribution!
It looks overall great, just a couple of small changes and should be good to go!

GordonBeeming committed:
refactor: replace author image URL logic with derived stores

Refactor the author image URL logic in multiple components to use 
derived stores for better reactivity and cleaner code. This change 
eliminates redundant code and ensures that the author image URL 
updates automatically when the user or commit data changes.

estib-vega commented:
Merging this.
Thanks so much again for the contribution (specially at late hours of the night in Australia!).
Keep those changes coming! :)
